### PR TITLE
docs: add status tables for all components

### LINKED ISSUE #15654: [docs] Add support tables for all components
### Describe the issue you're reporting

Most of the components (exporters, extensions, processors, receivers) have a table at the top of their README files describing the stability level, supported pipeline types and distribution in which they are included. Some of them don't though. I propose that I add the tables for the components without them:

Exporters:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/azuremonitorexporter
- https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/mezmoexporter

Extensions:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/awsproxy

Processors:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/deltatorateprocessor
- https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor
- https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/logstransformprocessor
- https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/metricsgenerationprocessor

Receivers:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/receivercreator
- https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/solacereceiver
- https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/sqlqueryreceiver

I figured I'd create an issue for this before jumping on the code and sending a PR.

### PR DESCRIPTION
**Pull request by andrzej-stencel**
Fixes #15654

How I got the details for the status tables for each component:
- Stability - this was the most tricky one; for some of the components, this information is not available. I put "unknown" for those and will kindly ask those components' maintainers to fill in the missing information.
  UPDATE: Thanks to all the code owners, all the missing stability level information is now filled in :tada: 
- Supported pipeline types - I tried to get that from the rest of the description; hopefully got it right. Please comment if you think something needs to be changed.
- Distributions - This was the easiest, I looked at the manifests for [core](https://github.com/open-telemetry/opentelemetry-collector-releases/blob/v0.63.1/distributions/otelcol/manifest.yaml) and [contrib](https://github.com/open-telemetry/opentelemetry-collector-releases/blob/v0.63.1/distributions/otelcol-contrib/manifest.yaml) distributions. Hopefully I got this right.

### CODE REVIEW

**Comment by astencel-sumo** [comment]
@pcwiese As the [code owner](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.63.0/.github/CODEOWNERS#L38) for the Azure Monitor Exporter, can you specify the [stability level](https://github.com/open-telemetry/opentelemetry-collector#stability-levels) of this component?

**Comment by astencel-sumo** [comment]
@dashpole @billmeyer @gjanco As the [code owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.63.0/.github/CODEOWNERS#L60) for the Mezmo Exporter, can you specify the [stability level](https://github.com/open-telemetry/opentelemetry-collector#stability-levels) of this component?

**Comment by astencel-sumo** [comment]
@Aneurysm9 @mxiamxia As the [code owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.63.0/.github/CODEOWNERS#L77) for the AWS Proxy extension, can you specify the [stability level](https://github.com/open-telemetry/opentelemetry-collector#stability-levels) of this component?

**Comment by astencel-sumo** [comment]
@owais @dmitryax As the [code owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.63.0/.github/CODEOWNERS#L128) for the Kubernetes Attributes Receiver, can you specify the [stability level](https://github.com/open-telemetry/opentelemetry-collector#stability-levels) of this component?

**Comment by astencel-sumo** [comment]
@jrcamp As the [code owner](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.63.0/.github/CODEOWNERS#L197) for the Receiver Creator component, can you specify the [stability level](https://github.com/open-telemetry/opentelemetry-collector#stability-levels) of this component?

**Comment by djaglowski** [comment]
> @djaglowski @mcardy As the [code owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.63.0/.github/CODEOWNERS#L206) for the Solace Receiver, can you specify the [stability level](https://github.com/open-telemetry/opentelemetry-collector#stability-levels) of this component?

I believe this one should be alpha.

**Comment by mcardy** [comment]
> @djaglowski @mcardy As the [code owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.63.0/.github/CODEOWNERS#L206) for the Solace Receiver, can you specify the [stability level](https://github.com/open-telemetry/opentelemetry-collector#stability-levels) of this component?

We have one more unit of work we would like to have before this component should be stable (est. late November/early December timeframe). Either alpha or beta is sufficient for now, but after that work the component should be stable and ready for production use. We have done significant internal validation, so will be confident in the stability of the receiver by December.

File: exporter/azuremonitorexporter/README.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Azure Monitor Exporter
 
+| Status                   |                       |
+| ------------------------ |-----------------------|
+| Stability                | unknown               |
```
Comment (pcwiese) [inline exporter/azuremonitorexporter/README.md:5]:
```suggestion
| Stability                | beta               |
```

File: extension/awsproxy/README.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # AWS Proxy
 
-##
+| Status                   |           |
+|--------------------------|-----------|
+| Stability                | unknown   |
```
Comment (Aneurysm9) [inline extension/awsproxy/README.md:5]:
```suggestion
| Stability                | beta   |
```

@mxiamxia let me know if you think this should be `stable` instead.  AFAIK the configuration has been stable for a while and I'm not aware of any plans to change it.

File: processor/k8sattributesprocessor/README.md [hunk 3]
```diff
@@ -1 +1,11 @@
+# Kubernetes Attributes Receiver
+
+| Status                   |                       |
+|--------------------------|-----------------------|
+| Stability                | unknown               |
```
Comment (dmitryax) [inline processor/k8sattributesprocessor/README.md:5]: Let's set `beta` here

File: receiver/receivercreator/README.md [hunk 4]
```diff
@@ -1,5 +1,11 @@
 # Receiver Creator
 
+| Status                   |                       |
+|--------------------------|-----------------------|
+| Stability                | unknown               |
```
Comment (dmitryax) [inline receiver/receivercreator/README.md:5]: Let's set `beta` here

**Comment by astencel-sumo** [comment]
Wow, I didn't expect so good responsiveness, but it seems we have stability levels settled for all the components. Thank you to everyone involved! :clap: :tada: :trophy: 

@bogdandrutu as the assignee, can you review this PR?

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
@astencel-sumo thanks for helping with this!

**Pull request merged**